### PR TITLE
doc: say that CMake is the default build system

### LINKED ISSUE #3089: Update CMake build documentation
The current documentation about building Charm++ with CMake still says that it is experimental, doesn't work on Windows, etc. This should be updated to reflect the current state of the build, seeing as it's the default when running `./build` in v6.11.0.

### PR DESCRIPTION
**Pull request by matthiasdiener**
and other various fixes

Fixes #3089.

### CODE REVIEW

**Comment by rbuch** [comment]
This should remove the "beta version" header from `CMakeLists.txt`, too.

**Comment by matthiasdiener** [comment]
> This should remove the "beta version" header from `CMakeLists.txt`, too.

It's still not really finished though - is there a better way we can express this?

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ cmake_minimum_required(VERSION 3.4)
 
 # Print warning
 message("#################################################################")
-message("# The CMake build system is currently a beta version.           #")
+message("# The CMake build system does not support all build options.    #")
```
Comment (rbuch) [inline CMakeLists.txt:17]: I think the current change sounds a little too negative, maybe something like this is a bit smoother:
```suggestion
message("# The CMake build system may not support advanced build options.    #")
```

Comment (matthiasdiener) [reply CMakeLists.txt]: That's a good idea, thanks!

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,7 +14,7 @@ cmake_minimum_required(VERSION 3.4)
 
 # Print warning
 message("#################################################################")
-message("# The CMake build system does not support all build options.    #")
+message("# The CMake build system may not support advanced build options.    #")
 message("# Please let us know if you experience any problems by opening  #")
 message("# an issue on our GitHub (https://github.com/UIUC-PPL/charm),   #")
 message("# with the below configuration information.                     #")
```
Comment (evan-charmworks) [inline CMakeLists.txt]: The alignment of the right border is broken now.

Comment (matthiasdiener) [reply CMakeLists.txt]: Thanks, fixed.

**Review by evan-charmworks — APPROVED** [review]

**Review by rbuch — APPROVED** [review]

**Pull request merged**